### ISSUE TITLE
Add Graphite and InfluxDB backends for Centreon Broker 3

### ISSUE THREAD

**Issue by ganoze** [centreon/centreon#4058]
Centreon Broker 3 is capable of writing performance data to Graphite and InfluxDB databases. However this is not yet supported in Centreon.

For Graphite, the changes will be minor. A new output type (*graphite*, just like *storage* or *sql*) must be added with the following fields.

  - **db_host** (string) : host of the Graphite database
  - **db_port** (integer) : port to connect to
  - **db_user** (string) : user name
  - **db_password** (string) : password associated with user name
  - **queries_per_transaction** (integer) : number of queries to group within a single transaction to increase performance
  - **metric_naming** : how to name entries for metrics. This string supports macros such as $METRIC$, $HOST$, $SERVICE$ and $INSTANCE$.
  - **status_naming** : how to name entries for statuses. Works just like *metric_naming*.

```xml
<centreonbroker>
  <output>
    <type>graphite</type>
    <name>my-graphite-output</name>
    <db_host>localhost</db_host>
    <db_port>80</db_port>
    <db_user>guest</db_user>
    <db_password>guest</db_password>
    <queries_per_transaction>1000</queries_per_transaction>
    <metric_naming>centreon.metrics.$METRIC$.$INSTANCE$.$HOST$.$SERVICE$</metric_naming>
    <status_naming>centreon.metrics.status.$INSTANCE$.$HOST$.$SERVICE$</status_naming>
  </output>
</centreonbroker>
```

For InfluxDB, things are similar to Graphite, an *influxdb* output type must be added along with the following fields.

  - **db_host**
  - **db_port**
  - **db_user**
  - **db_password**
  - **db_name** (string) : the database name
  - **metrics_timeseries** (string) : name of the metrics timeseries
  - **status_timeseries** (string) : name of the status timeseries

Then things become a little more complex. Users have to manually configure the timeseries layout by adding multiple **metrics_column** (for the metrics timeseries) or **status_column** (for the status timeseries) blocks. Any *influxdb* output can have an unlimited number of columns of any type. Each block must contain the following fields.

  - **name** (string) : name of the column (macros accepted)
  - **value** (string) : value of the column (macros accepted)
  - **is_tag** (boolean) : whether or not this column is a tag, default to false
  - **type** (string) : can be either *string* or *number* according to the column type, default to *string*

```xml
<centreonbroker>
  <output>
    <type>influxdb</type>
    <name>my-influxdb-output</name>
    <db_host>localhost</db_host>
    <db_port>8086</db_port>
    <db_user>guest</db_user>
    <db_password>guest</db_password>
    <db_name>centreon_influxdb</db_name>
    <metrics_timeseries>centreon.metrics.$METRIC$</metrics_timeseries>
    <metrics_column>
      <name>host</name>
      <value>$HOST$</value>
      <is_tag>true</is_tag>
    </metrics_column>
    <metrics_column>
      <name>value</name>
      <value>$VALUE$</value>
      <type>number</type>
    </metrics_column>
    <status_timeseries>centreon.metrics.status</status_timeseries>
    <status_column>
      <name>host</name>
      <value>$HOST$</value>
      <is_tag>true</is_tag>
    </status_column>
    <status_column>
      <name>value</name>
      <value>$VALUE$</value>
      <type>number</type>
    </status_column>
  </output>
</centreonbroker>
```

**Comment by dgarandel** [comment]
Stay just a little bug to write correct value to Influxdb.
I'm using centreon-broker-influxdb-3.0.3-4 (with centreon 3.4) and influxdb-1.1.1-1

Data's are correctly send to Influxdb but with a invalid timestamp format. Need to be in epoch format.

IE 👍 # curl -G 'http://localhost:8086/query?pretty=true' --data-urlencode "db=centreon" --data-urlencode 'q=SELECT "value" FROM "metrics.ethe-01-tc2-25434.Interface TenGigabitEthernet1/2/0.in"'

[...]
 [
                            "1970-01-01T00:00:01.483617798Z",
                            3.00917e+06
                        ]
[...]

INSERT STATUS,host=centeon-server,service=proc-ntpd,index_id=16 value=0.11 1480950952 is bad
INSERT STATUS,host=centeon-server,service=proc-ntpd,index_id=16 value=0.11 1480950952000000000 is good

in centreon-influxdb/centreon-broker/influxdb/src/influxdb10.cc I change these 2 sections by 

   if (!first)
     query_str.append(" ");
    query_str.append("$TIME$);
    query_str.append("000000000\n");     -> pad in epoch format

**Comment by ganoze** [comment]
Hi @dgarandel,

The timestamp format issue should be fixed with the upcoming 3.0.4 release of Centreon Broker. Modifications were made on timestamp format as discussed here : centreon/centreon-broker#85
